_client/src/Ui/Views/BaseView.cpp

```diff
@@ -3,64 +3,43 @@
 
 BaseView::BaseView(QWidget *parent) : QWidget(parent)
 {
-    setupUi();
+    setupUi(); // Call function to set up the user interface
 }
 
 BaseView::~BaseView() {}
 
 void BaseView::setupUi()
 {
-    // set main layout
-    m_mainLayout = new QVBoxLayout(this);
-    this->setLayout(m_mainLayout);
-
-    //  add buttons to button layout
-    m_buttonRow = new QHBoxLayout;
+    // Create buttons
     m_addButton = new QPushButton("Add", this);
     m_deleteButton = new QPushButton("Delete", this);
     m_editButton = new QPushButton("Edit", this);
 
-    m_buttonRow->addWidget(m_addButton);
-    m_buttonRow->addWidget(m_editButton);
-    m_buttonRow->addWidget(m_deleteButton);
-
-    // add spacer to button layout
-    m_additionalButtonSpace = new QSpacerItem(
+    // Create spacer item for the button layout
+    m_buttonRowSpacerItem = new QSpacerItem(
         ViewsStyles::ButtonsSpacer::WIDTH,
         ViewsStyles::ButtonsSpacer::HEIGHT,
         QSizePolicy::Expanding,
         QSizePolicy::Fixed);
-    m_buttonRow->addSpacerItem(m_additionalButtonSpace);
 
-    // add button layout to main layout
-    m_mainLayout->addLayout(m_buttonRow);
+    // Create a row layout for buttons and add spacer item
+    m_buttonRow = Utils::createLayout<QHBoxLayout>(this, {m_addButton, m_deleteButton, m_editButton, m_buttonRowSpacerItem});
 
-    // add data table to main layout
+    // Create a data table and add it to the main layout
     m_dataTable = new QTableWidget(this);
-    m_mainLayout->addWidget(m_dataTable);
-
-    // add label to additional info
-    m_status = new QLabel("Status: Ready", this);
-    m_additionalInfo = new QHBoxLayout;
-    m_additionalInfo->addWidget(m_status);
 
-    // add additional space under table in main layout
-    m_additionalUnderTableSpace = new QSpacerItem(
-        ViewsStyles::UnderTableSpacer::WIDTH,
-        ViewsStyles::UnderTableSpacer::HEIGHT,
+    // Create a status label and spacer item for additional info
+    m_status = new QLabel("Status: ready");
```

Comment:
Who is managing dynamic memory allocation?

Reply:
I totally forgot about destructors